src/jsonschema.rs

```diff
@@ -128,34 +136,42 @@ impl Tag {
                         &self.object.pattern_properties,
                     ) {
                         (Some(AdditionalProperties::Object(add)), Some(pat)) => {
-                            let value = ast::Tag::new(
-                                ast::Type::Union(ast::Union::new(vec![
-                                    add.type_into_ast(),
-                                    pat.type_into_ast(),
-                                ])),
-                                None,
-                                false,
-                            );
+                            let mut vec: Vec<ast::Tag> = Vec::new();
+                            vec.push(add.type_into_ast());
+                            for value in pat.values() {
```

Comment:
This should work as well and might be slightly more efficient:

```rust
vec.extend(pat.values().map(|v| v.type_into_ast()))
```

But then again the more explicit for loop is probably easier to understand.